src/init/features/frameworks/index.ts

```diff
@@ -26,32 +26,40 @@ export async function doSetup(setup: any, projectId: string): Promise<void> {
 
   logBullet("First we need a few details to create your backend.");
 
-  await promptOnce(
-    {
-      name: "serviceName",
+  const location = await promptOnce({
+    name: "region",
+    type: "list",
+    default: DEFAULT_REGION,
+    message:
+      "Please select a region " +
+      `(${clc.yellow("info")}: Your region determines where your backend is located):\n`,
+    choices: ALLOWED_REGIONS,
+  });
+
+  logSuccess(`Region set to ${location}.\n`);
+
+  let backendId: string;
+  while (true) {
```

Comment:
Why?

Reply:
This is effectively an infinite loop to ask for user input when they give an invalid name for the backend, e.g.

```
$ firebase --project danielylee-test-6  backends:create
i  First we need a few details to create your backend.
? Please select a region (info: Your region determines where your backend is located):
 us-central1 (Iowa)
✔  Region set to us-central1.

? Create a name for your backend [1-30 characters] backend
⚠  Backend with id backend already exists in us-central1
? Create a name for your backend [1-30 characters] backend
⚠  Backend with id backend already exists in us-central1
? Create a name for your backend [1-30 characters] (acme-inc-web)
```

Reply:
consider adding a `err.status === 409` then so this is explicit

Reply:
Instead of relying on 409, the CLI make a GET request to check if backend exists early in the onboarding. This prevents cases where users goes through all the hurdles to create a new GitHub connection only to have their backend failed to be created because of a bad name.

I think there are other flows we could've written to use 409 more dynamically, but I think the current implementation is consistent with how the console would work.